api/src/main/scala/Geoprocessing.scala

```diff
@@ -18,10 +18,122 @@ trait Geoprocessing extends Utils {
     * @param   input  The InputData
     * @return         A histogram of results
     */
-  def getRasterGroupedCount(input: InputData): Result = {
+  def getRasterGroupedCount(input: InputData): ResultInt = {
     val aoi = createAOIFromInput(input)
     val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
-    Result(rasterGroupedCount(rasterLayers, aoi))
+    ResultInt(rasterGroupedCount(rasterLayers, aoi))
+  }
+
+  /**
+    * For an InputData object, return a histogram of raster grouped average
+    * results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results
+    */
+  def getRasterGroupedAverage(input: InputData): ResultDouble = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
+    val targetRaster = input.targetRaster
+    val targetLayer = targetRaster match {
```

Comment:
Since this is the only use of `targetRaster`, consider combining this with the line above:

```scala
val targetLayer = input.targetRaster match {
```